src/TabCompletion/AutoCompleter.php

```diff
@@ -23,6 +23,84 @@ class AutoCompleter
     /** @var Matcher\AbstractMatcher[] */
     protected $matchers;
 
+    /**
+     * The set of characters which separate completeable 'words', and
+     * therefore determines the precise $input word for which completion
+     * candidates should be generated (noting that each candidate must
+     * begin with the original $input text).
```

Comment:
You're welcome.  This whole thing ended up as a bit of a learning spiral for me -- every time I got to grips with something, I had to revisit some of my previous work.  I really hadn't anticipated needing to understand things to this extent, and eventually realised that I'd approached some aspects almost completely backwards.  Hopefully the docs will help other people to avoid the same detours in future.